pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -172,221 +168,192 @@ func (c *MizarArktosNetworkController) syncNetwork(eventKeyWithType KeyWithEvent
-	// Get json file in bytes format
-	manifestData, err := GetCRDVpcOrSubnetSpec(defaultTemplatePath, vpcOrSubnetName, tenant)
+	// Create Subnet object
+	subnetSpec, err := generateSubnetSpec(vpc, subnet, vpcIpStart)
 	if err != nil {
-		klog.Errorf("At (%s) get JSON spec for CRD (%s) in error: (%v)", defaultTemplatePath, vpcOrSubnetName, err)
+		klog.Errorf("Error getting Subnet %s spec. Error: %v", subnet, err)
 		return err
 	}
-
-	// Create VPC or Subnet object
-	err = createUnstructuredObject([]byte(manifestData), vpcOrSubnetName, discoveryClient, dynamicClient)
-	if err != nil {
-		klog.Errorf("Create CRD object (%s) in error: (%v)", vpcOrSubnetName, err)
+	err = createUnstructuredObject(subnetSpec, discoveryClient, dynamicClient)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Errorf("Error creating Subnet %s for VPC %s. Error: %v", subnetSpec, vpc, err)
 		return err
 	}
-	klog.V(3).Infof("Create CRD object: (%s) successfully", vpcOrSubnetName)
 
 	return nil
 }
 
-func createUnstructuredObject(data []byte, vpcOrSubnetName string, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
+func createUnstructuredObject(data []byte, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
 	unstructuredObj := &unstructured.Unstructured{}
 	decUnstructured := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
 
 	// Get GVK(Group Version Kind)
 	_, gvk, err := decUnstructured.Decode(data, nil, unstructuredObj)
 	if err != nil {
-		klog.Errorf("Getting GVK in error (%v).", err)
+		klog.Errorf("Getting GVK for %v/%v result in error (%v).", unstructuredObj.GetTenant(), unstructuredObj.GetName(), err)
 		return err
 	}
-	klog.V(4).Infof("Get Name : (%s) and GVK: (%s)", unstructuredObj.GetName(), gvk.String())
 
 	// Get mapping from GVK for GVR (Group Version Resource) used by dynamic client resource
 	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
-	klog.V(4).Infof("Name: %s - GVK group kind : (%v) - GVK version: (%v)", unstructuredObj.GetName(), gvk.GroupKind(), gvk.Version)
 	mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
 
-	klog.V(4).Infof("Name: %s - GVK group kind : (%s) - GVK version: (%s)", unstructuredObj.GetName(), unstructuredObj.GetKind(), unstructuredObj.GetAPIVersion())
 	if err != nil {
 		klog.Errorf("Get mapping between GVK and GVR in error (%v).", err)
 		return err
 	}
 
-	klog.V(4).Infof("Name: %s - get mapping scope name: (%s) - meta RESTScopeNameNamespace: (%s)", unstructuredObj.GetName(), mapping.Scope.Name(), meta.RESTScopeNameNamespace)
-
 	// Create dynamic client resource
 	var dynamicClientResource dynamic.ResourceInterface
 	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
 		// namespaced resources should specify the namespace under tenant
+		// VPC/Subnet is tenant unaware
```

Comment:
should this be namespace unaware ?  vpc is tenant scope resources

Reply:
> should this be namespace unaware ? vpc is tenant scope resources

As I mentioned earlier, createUnstructuredObject() function needs further changes. It can be done later. Right now, I am limiting the scope of this PR.